src/backend/resourcemanager/requesthandler.c

```diff
@@ -627,7 +627,11 @@ bool handleRMSEGRequestIMAlive(void **arg)
 	fts_client_ip_len = strlen(fts_client_ip);
 	inet_aton(fts_client_ip, &fts_client_addr);
 	fts_client_host = gethostbyaddr(&fts_client_addr, 4, AF_INET);
-	Assert(fts_client_host != NULL);
+	if (fts_client_host == NULL)
+	{
+		elog(WARNING, "Failed to reverse DNS lookup for ip %s.", fts_client_ip);
+		return true;
+	}
 
```

Comment:
If it is easy to modify we could use the replacement of gethostbyaddr().
By the way, why need to quit after "reverse dns lookup" failure. Should not
an IP address work also?

DESCRIPTION
       The gethostbyname*() and gethostbyaddr*()  functions  are  obsolete.   Applications
       should use getaddrinfo(3) and getnameinfo(3) instead.